app/questionnaire/questionnaire_store_updater.py

```diff
@@ -330,10 +378,12 @@ def update_progress_for_dependant_sections(self) -> None:
                 section_id != self._current_location.section_id
                 or list_item_id != self._current_location.list_item_id
             ):
-                self.update_section_status(
-                    is_complete=False,
-                    section_id=section_id,
-                    list_item_id=list_item_id,
+                dependent = DependentSection(section_id, list_item_id, None)
+                if dependent in self.dependent_sections:
+                    self.dependent_sections.remove(dependent)
```

Comment:
I think this warrants a comment, as it is not obvious why we are removing.

Reply:
Added a comment